### PR TITLE
Increase the license options.

### PR DESCRIPTION
**Pull request by Lokathor**
Per https://github.com/Frommi/miniz_oxide/issues/66, this adds the `Zlib` license as a possible license for users of the crate. While we're at it, it's probably best to also add the `Apache-2.0` license as an option.

I expect this to be fairly uncontroversial, but this would need sign off from past code contributors (alphabetical order):
* [x] alexcrichton
* [ ] arbitrix
* [x] ctjhoa
* [x] dholbert
* [x] dingelish
* [x] eminence
* [x] FauxFaux
* [x] fhanau
* [x] Frommi
* [x] golddranks
* [x] ignatenkobrain
* [x] jonas-schievink
* [x] matklad
* [x] oyvindln
* [x] RReverser
* [x] Shnatsel
* [x] srijs
* [x] vlad20012
* [ ] wehlutyk

If @Frommi is still on board with the idea then we can start pinging the rest of people.

### CODE REVIEW

**Comment by Lokathor** [comment]
Frommi checked their box, so I'll ping everyone else:
@arbitrix
@ctjhoa
@dholbert
@dingelish
@eminence
@FauxFaux
@fhanau
@golddranks
@ignatenkobrain
@jonas-schievink
@matklad
@oyvindln
@RReverser
@Shnatsel
@srijs
@vlad20012
@wehlutyk

**Comment by dingelish** [comment]
Signed-off by Yu Ding @dingelish 
:p

**Comment by ignatenkobrain** [comment]
Fine with me. But please add texts of those licenses into the tree.

**Comment by Lokathor** [comment]
Ah, can do.

**Comment by ctjhoa** [comment]
I'm ok too

**Comment by matklad** [comment]
I agree to adding Zlib and Apache 2.0 licenses

**Comment by FauxFaux** [comment]
I don't even remember contributing, which probably means that my contribution is too small for you to even ask.

But, yes, please do.

**Comment by golddranks** [comment]
I agree!

**Comment by Shnatsel** [comment]
I license past and future contributions under Zlib OR Apache-2.0 or MIT license, allowing licensees to chose any one at their option.

**Comment by jonas-schievink** [comment]
I'm fine with this too

**Comment by oyvindln** [comment]
I'm fine with this.

**Comment by Lokathor** [comment]
Apologies, there was apparently a copy/paste mixup and @alexcrichton wasn't initially on the list, so you get a ping too.

**Comment by srijs** [comment]
👍 fine by me

**Comment by eminence** [comment]
I agree to adding Zlib and Apache 2.0 licenses

**Comment by dholbert** [comment]
I've only contributed [one commit](https://github.com/Frommi/miniz_oxide/commit/f398d88259e957c5360f3689b55007d9de50ec6b) which was a 1-character typo-fix, so I'm arguably not really a code contributor. :) But, setting that aside: sure, I'm fine with this license change.

**Comment by fhanau** [comment]
I also agree to adding the licenses.

**Comment by vlad20012** [comment]
I agree to adding Zlib and Apache 2.0 licenses

File: LICENSE-APACHE.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Comment (vlad20012) [inline LICENSE-APACHE.md:179]: Looks like appendix can be removed. E.g. it's omitted in the rust repo https://github.com/rust-lang/rust/blob/master/LICENSE-APACHE

Comment (Lokathor) [reply LICENSE-APACHE.md:179]: good catch!

**Comment by RReverser** [comment]
No problem from my side.

**Comment by alexcrichton** [comment]
I agree to this change of licenses.

**Comment by Lokathor** [comment]
I made a gaffe and would like to apologize to everyone.

The PR had been set to affect the `miniz_oxide_c_api`. I **intended** to change things for the `miniz_oxide` crate. I was not clear on the repo organization at the time (the intended crate is in a sub-directory), and so I altered the wrong Cargo.toml and added the new licenses all at the repository root.

I have pushed the proper changes so that this PR now applies to **just** the `miniz_oxide` crate.

Since the relicense has been signed off on by everyone who has touched that crate specifically:
* @Frommi I ask that you merge this PR as is, and please push out a release to crates.io some time soon.
* I'll open a second PR for the C API crate to also get the re-license treatment, with a ping to just the one last contributor we'd need a sign off from for the C api to be also re-licensed.

**Comment by oyvindln** [comment]
Okay, so the remaining two contributors, @arbitrix and @wehlutyk contributed to the C API part only so I think this should be fine to merge, looks like both have not been active lately. (@arbitrix [contribution](https://github.com/Frommi/miniz_oxide/commit/c54f10d08032bf017676957f05d84aa63dfbef39) was just removing a line, so I don't know if that is something that counts as needing approval either.)

**Comment by Lokathor** [comment]
@oyvindln i resolved the merge conflicts that came up recently. Can you merge and publish yourself? Or do we need to keep waiting for @Frommi ?

**Comment by oyvindln** [comment]
I can do that, may wait a bit with publishing unless it's urgent, would like to get #97 in.

**Comment by Lokathor** [comment]
Understandable. I'll subscribe to that PR too.

Thanks for getting to this!

**Comment by arbitrix** [comment]
I agree to this change of licenses.

**Pull request merged**